java-squid/src/main/java/org/sonar/java/se/checks/UnclosedResourcesCheck.java

```diff
@@ -61,76 +59,55 @@
 @SqaleConstantRemediation("5min")
 public class UnclosedResourcesCheck extends SECheck implements JavaFileScanner {
 
-  abstract static class SingleTreeNodeVisitor extends BaseTreeVisitor {
-
-    private static final String JAVA_IO_CLOSEABLE = "java.io.Closeable";
-    private static final String[] IGNORED_CLOSEABLE_SUBTYPES = {
-      "java.io.ByteArrayOutputStream",
-      "java.io.ByteArrayInputStream",
-      "java.io.StringReader",
-      "java.io.StringWriter",
-      "java.io.CharArrayReader",
-      "java.io.CharArrayWriter",
-      "com.sun.org.apache.xml.internal.security.utils.UnsyncByteArrayOutputStream"
-    };
-
-    ProgramState programState;
-
-    protected SingleTreeNodeVisitor(ProgramState programState) {
-      this.programState = programState;
-    }
-
-    @Override
-    protected void scan(Tree tree) {
-      // Cut recursive processing
-    }
-
-    @Override
-    protected void scan(List<? extends Tree> trees) {
-      // Cut recursive processing
-    }
-
-    @Override
-    protected void scan(ListTree<? extends Tree> listTree) {
-      // Cut recursive processing
-    }
+  private static enum States {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this redundant "static" qualifier. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2786)